### PR TITLE
optimize: optimize seata-server log pattern and support for colored log.

### PR DESCRIPTION
**Pull request by wangliang1986**
### Ⅰ. Describe what this PR did
seata-server端的日志相关优化：
1、日志格式调整，使日志更整齐。
2、控制台打印日志时，部分文字带颜色。
3、日志文件名拼接端口号，防止集群时部分日志写入失败。
4、日志文件按日志级别分开：txc.${port}.all.log、txc.${port}.warn.log、txc.${port}.error.log
5、历史日志文件，放入子文件夹/history中，且历史文件后缀名改为.gz

相关功能及日志格式，从spring-boot.jar中复制过来的，我只是搬运工，哈哈。

### **优化后的效果如下：**
**控制台输出：**
after-console
**日志文件输出：**
after-file

### 优化前的效果如下：
_控制台输出：_
before-console
_日志文件输出：_
before-file

### CODE REVIEW

**Comment by wangliang1986** [comment]
> 
> 
> @wangliang1986 please explain the display comparison before and after modification.

The comparison chart is in the description, please check it.

File: server/src/main/resources/logback.xml [hunk 1]
```diff
@@ -16,11 +16,20 @@
   -->
 
 <configuration scan="true" scanPeriod="60 seconds" debug="false">
+    <!-- Copied from spring-boot.jar -->
+    <conversionRule conversionWord="clr" converterClass="io.seata.server.logging.logback.ColorConverter" />
+    <conversionRule conversionWord="wex" converterClass="io.seata.server.logging.logback.WhitespaceThrowableProxyConverter" />
+    <conversionRule conversionWord="wEx" converterClass="io.seata.server.logging.logback.ExtendedWhitespaceThrowableProxyConverter" />
+    <conversionRule conversionWord="pid" converterClass="io.seata.server.logging.logback.PidConverter" />
+    <property name="CONSOLE_LOG_PATTERN" value="%clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr(%5p) %clr(%pid){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n%wEx"/>
+    <property name="FILE_LOG_PATTERN" value="%d{yyyy-MM-dd HH:mm:ss.SSS} %5p %pid --- [%t] %-40.40logger{39} : %m%n%wEx"/>
```
Comment (jsbxyyx) [inline server/src/main/resources/logback.xml:25]: logger length original is 56, now is 39, is there a difference?

Comment (wangliang1986) [reply server/src/main/resources/logback.xml:25]: More info added before, so 56 to 39.
Avoid the log message can't look on the small screen.

File: server/src/main/resources/logback.xml [hunk 2]
```diff
@@ -16,39 +16,91 @@
...
-        <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
-            <Pattern>%date{yyyy-MM-dd HH:mm:ss.SSS} %-5level[%thread]%logger{56}.%method:%L -%msg%n</Pattern>
+    <!-- Copied from spring-boot.jar -->
+    <conversionRule conversionWord="clr" converterClass="io.seata.server.logging.logback.ColorConverter" />
+    <conversionRule conversionWord="wex" converterClass="io.seata.server.logging.logback.WhitespaceThrowableProxyConverter" />
+    <conversionRule conversionWord="wEx" converterClass="io.seata.server.logging.logback.ExtendedWhitespaceThrowableProxyConverter" />
+    <property name="CONSOLE_LOG_PATTERN" value="%clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr(%5p) %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n%wEx"/>
+    <property name="FILE_LOG_PATTERN" value="%d{yyyy-MM-dd HH:mm:ss.SSS} %5p --- [%t] %-40.40logger{39} : %m%n%wEx"/>
+
+    <property name="LOG_HOME" value="./logs"/>
```
Comment (jsbxyyx) [inline server/src/main/resources/logback.xml:29]: original value is ${user.home}/logs/seata

File: server/src/main/java/io/seata/server/logging/logback/ApplicationPid.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+package io.seata.server.logging.logback;
+
+import java.lang.management.ManagementFactory;
+
+/**
+ * An application process ID.
+ *
+ * @author wang.liang
+ */
+public class ApplicationPid {
```
Comment (jsbxyyx) [inline server/src/main/java/io/seata/server/logging/logback/ApplicationPid.java:25]: The class not used?

**Pull request merged**